sklearn/metrics/ranking.py

```diff
@@ -255,9 +269,41 @@ def _binary_roc_auc_score(y_true, y_score, sample_weight=None):
                                         sample_weight=sample_weight)
         return auc(fpr, tpr, reorder=True)
 
-    return _average_binary_score(
-        _binary_roc_auc_score, y_true, y_score, average,
-        sample_weight=sample_weight)
+    y_type = type_of_target(y_true)
+    y_true = check_array(y_true, ensure_2d=False)
+    y_score = check_array(y_score, ensure_2d=False)
+
+    if y_type == "multiclass" or (y_type == "binary" and
```

Comment:
Not sure if this is the best way to handle the case mentioned (where `y_true` appears binary but `y_score` has more than 2 classes)